ui/selectors/assets.ts

```diff
@@ -926,6 +931,106 @@ export const selectBalanceChangeBySelectedAccountGroup = (
     },
   );
 
+/**
+ * Creates an enabledNetworkMap from all non-test networks for balance calculations.
+ * This selector combines EVM and non-EVM mainnet networks (excluding testnets and custom testnets)
+ * and formats them into the enabledNetworkMap structure expected by calculateBalanceForAllWallets.
+ *
+ * @param state - Redux state containing network configurations.
+ * @returns EnabledNetworkMap with all non-test networks enabled across all namespaces.
+ */
+const selectAllMainnetNetworksEnabledMap = createSelector(
+  [getNonTestNetworks],
+  (nonTestNetworks) => {
+    const enabledNetworkMap: Record<string, Record<string, boolean>> = {};
+
+    nonTestNetworks.forEach((network) => {
+      const { caipChainId } = network;
+      const { namespace, reference } = parseCaipChainId(caipChainId);
+
+      if (!enabledNetworkMap[namespace]) {
+        enabledNetworkMap[namespace] = {};
+      }
+
+      // Fix: Convert reference to proper format for calculateBalanceForAllWallets
+      if (namespace === KnownCaipNamespace.Eip155) {
+        // For EVM chains, use hex format (e.g., "1" → "0x1")
+        const chainIdHex = toHex(reference);
+        enabledNetworkMap[namespace][chainIdHex] = true;
+      } else {
+        // For non-EVM chains, use full CAIP chainId as key
+        enabledNetworkMap[namespace][caipChainId] = true;
+      }
+    });
+
+    return enabledNetworkMap;
+  },
+);
+
+/**
+ * Calculates total balance across all enabled networks for the selected account group.
+ *
+ * @param state - Redux state containing all required controller states for balance calculation.
+ * @returns The total balance in user currency for the selected account group, or 0 if no group is selected.
+ */
+export const selectAccountGroupBalanceForEmptyState = createSelector(
```

Comment:
Nice selector, my worry is that we've found balances to be highly volatile (a lot of pieces of state used spread throughout the app + currency rates can fluctuate).

Since we only need to check for a non-zero balance, what are your thoughts on changing the return type primitive from `number` to `boolean`? Although this will still cause the selector to recompute from small changes, the result and component tree will not change.

Reply:
@georgewrmarshall aha understood! No worries on reusing existing selectors, I'll discuss with the @MetaMask/metamask-assets to investigate our selectors for this usecase!